azkaban-web-server/src/main/java/azkaban/imagemgmt/servlets/ImageTypeServlet.java

```diff
@@ -44,15 +47,19 @@
  * This servlet exposes the REST APIs such as create, get etc. for image type. Below are the
  * supported APIs. Create Image Type API: POST /imageTypes?session.id=? --data @payload.json Get
  * Image Type API: GET /imageTypes?session.id=?&imageType=? GET /imageTypes/{id}?session.id=?
+ * GET /imageTypes/{imageType}?sessionid=? GET /imageTypes/{imageType}?proxyUser=?&session.id=?
```

Comment:
What prevents someone from specifying an arbitrary `proxyUser` or will there be some additional checks ?

Reply:
My understanding is security through obscurity for native calls to the API itself. Through Jenkins, we will be constructing the API call from the User who clicks the button (as the User is passed from Jenkins to the Python job). I hope that answers your question?

Reply:
'the User who clicks the button' will be the user specified in the 'proxyUser' parameter, is that correct ?
I would recommend making the security/privileges for this more robust as otherwise this could be open to accidental abuse and would be a more serious concern if this paradigm was extended to more APIs. 
One way of doing this could be to have an allow-list (perhaps in config files) that grants these proxy/impersonation privileges to a specific set of users (which would be the user Jenkins uses and the one we fetch the session-id for). 

In either case I think this warrants a broader discussion with the team (we can do this offline)

Reply:
> 'the User who clicks the button' will be the user specified in the 'proxyUser' parameter, is that correct ?

Yes, that is correct.

The code as it is written now, will make a request from Jenkins using the az_test user, which I believe is an admin on all servers. Therefore, at least for Jenkins, I don't think the proxyUser logic will be utilized, and another check will be necessary to make sure the user passed from Jenkins is authorized to make changes. I believe the proxyUser logic is solely for calls into the API by users. @jakhani, can you confirm?

Reply:
@sshardool @ypadron-in ,
Logic will be like:
Case 1: Proxy user param is not passed:
Check that logged in user has permission on Image type. If yes, then return image type dto. If not then throw exception.
Case 2: Proxy user param is passed:
Check that logged in user has permission on Image type. If yes, then check that proxy user has permission on image type. If yes then return image type DTO. If not then throw exception.

Logged in user can be Azkaban admin and willing to verify whether proxy user can access any image type or not. Here proxy user can be either Azkaban admin or image type owner.